packages/agent-utils/src/agent/ChatMessageBuilder.ts

```diff
@@ -38,7 +38,7 @@ export class ChatMessageBuilder {
       return [
         ChatMessageBuilder.functionCallResult(funcName, `Due to it's size, the result is stored in variable \${${varName}} (total length: ${result.length}).
 To read different parts of the stored variable, use the function readVariable("\${${varName}}", start, count)`),
-        ChatMessageBuilder.functionCall("readVariable", { name: varName, start: 0, count: threshold}),
+        ChatMessageBuilder.functionCall("readVariable", { name: `\${${varName}}`, start: 0, count: threshold}),
```

Comment:
Why do we add ${} for all var names? Don't var names work fine?

Reply:
This is needed to help the LLM know how to pass variable names into other functions. Without this, it sees `readVariable(name: "var-name", ...)`, and then later it will try `writeFile(data: "var-name", ...)`.